daemon/src/supervisor.rs

```diff
@@ -98,3 +111,60 @@ where
 pub struct Stopped<R> {
     pub reason: R,
 }
+
+/// Return the metrics tracked by this supervisor.
+pub struct GetMetrics;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use xtra::Actor as _;
+
+    #[tokio::test]
+    async fn supervisor_tracks_spawn_metrics() {
+        let (supervisor, address) =
+            Actor::new(|supervisor| RemoteShutdown { supervisor }, |_| true);
+        let (supervisor, task) = supervisor.create(None).run();
+        tokio::spawn(task);
+
+        let metrics = supervisor.send(GetMetrics).await.unwrap();
```

Comment:
Why does sending a message spawn a new actor?

Reply:
It doesn't! What spawns the actor initially is the `started` lifecycle hook of the supervisor:

https://github.com/itchysats/itchysats/blob/b412ac01c5e5f8387210f39fb02e995414a13626/daemon/src/supervisor.rs#L61-L63